protos/flyteidl/admin/description_entity.proto

```diff
@@ -0,0 +1,104 @@
+import "flyteidl/admin/common.proto";
+
+// DescriptionEntity contains detailed description for the task/workflow/launch plan.
+// Documentation could provide insight into the algorithms, business use case, etc.
+message DescriptionEntity {
+  // One-liner overview of the entity.
+  string short_description = 1;
+  // Full user description with formatting preserved.
+  LongDescription long_description = 2;
+  // Optional link to source code used to define this entity.
+  SourceCode source_code = 3;
+  // User-specified tags. These are arbitrary and can be used for searching
+  // filtering and discovering tasks.
+  repeated string tags = 4;
+}
+
+// Full user description with formatting preserved. This can be rendered
+// by clients, such as the console or command line tools with in-tact
+// formatting.
+message LongDescription {
+  oneof content {
+    // long description - no more than 4KB
+    string value = 1;
+    // if the description sizes exceed some threshold we can offload the entire
+    // description proto altogether to an external data store, like S3 rather than store inline in the db
+    string uri = 2;
+  }
+
+  enum DescriptionFormat {
+    UNKNOWN = 0;
+    MARKDOWN = 1;
+    HTML = 2;
+    // python default documentation - comments is rst
+    RST = 3;
+  }
+  // Format of the long description
+  DescriptionFormat format = 3;
+  // Optional link to an icon for the entity
+  string icon_link = 4;
+}
+
+// Link to source code used to define this entity
+message SourceCode {
+  string link = 1;
+}
+
+// Encapsulation of fields that uniquely identifies a Flyte description entity
+message DescriptionEntityIdentifier {
+  // Identifies the specific type of resource that this identifier corresponds to.
+  core.ResourceType resource_type  = 1;
+
+  // Name of the project the resource belongs to.
+  string project              = 2;
+
+  // Name of the domain the resource belongs to.
+  // A domain can be considered as a subset within a specific project.
+  string domain               = 3;
+
+  // User or system provided value for the resource.
+  string name                 = 4;
+}
+
+// Represents a list of DescriptionEntities returned from the admin.
+// See :ref:`ref_flyteidl.admin.DescriptionEntity` for more details
+message DescriptionEntityList {
+  // A list of DescriptionEntities returned based on the request.
+  repeated DescriptionEntity descriptionEntities = 1;
+
+  // In the case of multiple pages of results, the server-provided token can be used to fetch the next page
+  // in a query. If there are no more results, this value will be empty.
+  string token                = 2;
+}
+
+// Represents a request structure to retrieve a list of DescriptionEntities.
+// See :ref:`ref_flyteidl.admin.DescriptionEntity` for more details
+message DescriptionEntityListRequest {
+  // Indicates the node execution to filter by.
+  // +required
+  DescriptionEntityIdentifier description_entity_id = 1;
```

Comment:
I don't get why are we passing the identifier in the list request?

Reply:
iiuc - this will filter based on the fields that are filled in in the identifier.  so if you have resource type and project, then it'll do the list on those.  plus whatever extra filters there are.

Reply:
Listing today is done by `admin.ResourceListRequest` or `admin.NamedEntityIdentifierListRequest`.  i think both of these don't differentiate between tasks and workflows, but when listing descriptions we do.  so i think that's why this is here, but yeah @pingsutw can you confirm?

Reply:
ic... maybe the name of the message need to change then? as it stands, it's not really an identifier for the description entity.. it's not versioned

Reply:
actually yeah that made me realize... the `DescriptionEntity` object is the only kinda high level user facing object that doesn't carry it's ID with it.  Is that okay?  like if a user calls `GetDescriptionEntity`, the description returned has lost its connection with the root entity.  is this okay?  @EngHabu

Reply:
`s/it's/its/`

That's a good point... I think maybe we should have a DescriptionEntity wrapper (like those Admin wrappers for tasks), that has the underlying DescriptionEntity object + ID ?

Reply:
(edited)

Reply:
yeah okay, we're going to add id.